ouroboros-consensus-test/bench-mempool/Bench/Consensus/Mempool/Options.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE NumericUnderscores         #-}
+
+module Bench.Consensus.Mempool.Options (
+    -- * Custom options
+    customBenchIngredients
+  , customOpts
+    -- * Running commands
+  , Ns (..)
+    -- * Backing store selector
+  , BSS
+  , fromOptionBSS
+  ) where
+
+import           Bench.Consensus.Mempool.Params (defaultInMemoryBSS,
+                     defaultLMDB_BSS)
+import           Data.Char (toLower)
+import           Data.Proxy (Proxy (..))
+import           Options.Applicative (metavar)
+import           Ouroboros.Consensus.Storage.LedgerDB.Init
+                     (BackingStoreSelector)
+import           Test.Tasty (includingOptions)
+import           Test.Tasty.Bench (benchIngredients)
+import           Test.Tasty.Ingredients (Ingredient)
+import           Test.Tasty.Options (IsOption (..), OptionDescription (..),
+                     mkOptionCLParser, safeRead)
+
+{-------------------------------------------------------------------------------
+  Custom options
+-------------------------------------------------------------------------------}
+
+customBenchIngredients :: [Ingredient]
+customBenchIngredients = includingOptions customOpts : benchIngredients
+
+customOpts :: [OptionDescription]
+customOpts = [Option (Proxy :: Proxy BSS), Option (Proxy :: Proxy Ns)]
+
+{-------------------------------------------------------------------------------
+  Running commands
+-------------------------------------------------------------------------------}
+
+-- | Number of commands to run in the mempool benchmarks
+--
+-- Note: as this is a list of 'Int', the benchmarks will be run for each of
+-- these values.
+newtype Ns = Ns { getNs :: [Int] }
```

Comment:
Could we give this type a more descriptive name? Eg. `NumberOfCommandsToRun`, or `CommandsLengths`?

Reply:
https://github.com/input-output-hk/ouroboros-network/pull/4476#discussion_r1165395730